src/DomDocuments/BaseDocument.php

```diff
@@ -156,13 +156,24 @@ protected function appendPerformanceTypeFields(\DOMElement $element, $object): v
      * Use this instead of createElement().
      *
      * @param string $tag
-     * @param string $textContent
+     * @param string|null $textContent
+     * @param $object
+     * @param array|null $attributes
      * @return \DOMElement
      */
-    final protected function createNodeWithTextContent(string $tag, string $textContent): \DOMElement
+    final protected function createNodeWithTextContent(string $tag, ?string $textContent, $object = null, array $attributes = null): \DOMElement
```

Comment:
Again, the last argument can default to `[]`.

Reply:
Done